source/agora/consensus/ValidatorSet.d

```diff
@@ -319,6 +319,33 @@ public class ValidatorSet
         }
     }
 
+    /***************************************************************************
+
+        Gets the number of valid validators at the block height.
+
+        Params:
+            block_height = the height of proposed block
+
+    ***************************************************************************/
+
+    public ulong getValidatorCount (ulong block_height) @safe nothrow
+    {
+        ulong validator_count = 0;
+        try
+        {
+            ulong height = (block_height >= Enrollment.ValidatorCycle) ?
+                block_height - Enrollment.ValidatorCycle + 1 : 0;
+            () @trusted {
+                validator_count = this.db.execute("SELECT count(*) FROM validator_set WHERE enrolled_height >= ?", height).oneValue!ulong;
+            }();
+        }
+        catch (Exception ex)
+        {
+            log.error("Database operation error {}", ex);
```

Comment:
We will need to handle this better in the future. But it might be left for a future PR.